cli.go

```diff
@@ -52,9 +54,19 @@ func (h *CLIHandler) Auth(c *api.Client, m map[string]string) (*api.Secret, erro
 		callbackHost = defaultCallbackHost
 	}
 
+	callbackMethod, ok := m["callbackmethod"]
+	if !ok {
+		callbackMethod = defaultCallbackMethod
+	}
+
+	callbackport, ok := m["callbackport"]
+	if !ok {
+		callbackport = defaultCallbackPort
```

Comment:
I think this also needs to take into account whether `port` was set in order to maintain the existing behavior.

For example, this command will fail with the PR as is: `vault login -method=oidc role=test port=8400`. In the released version the redirect_uri is set to 8400, but with the PR it's now getting the default of 8250.

Reply:
Added a fix that will set `callbackport` to the value of `port` if no explicit `callbackport` option is passed.